tests/unit/test_ecies.cpp

```diff
@@ -114,6 +114,41 @@ TEST_F(ECIESTests, testWithSalts)
     EXPECT_EQ(testString, result);
 }
 
+void testGetEcPublicKeyOctetLength(AsymmetricPublicKey ephKey, AsymmetricKeypair key)
+{
+    size_t key_len_compressed_actual = ephKey.toECPoint(openssl::EllipticCurvePointConversionForm::compressed).size();
+    size_t key_len_compressed_calculated =
+        getEcPublicKeyOctetLength(key, openssl::EllipticCurvePointConversionForm::compressed);
+    EXPECT_EQ(key_len_compressed_actual, key_len_compressed_calculated);
+
+    size_t key_len_hybrid_actual = ephKey.toECPoint(openssl::EllipticCurvePointConversionForm::hybrid).size();
+    size_t key_len_hybrid_calculated =
+        getEcPublicKeyOctetLength(key, openssl::EllipticCurvePointConversionForm::hybrid);
+    EXPECT_EQ(key_len_hybrid_actual, key_len_hybrid_calculated);
+
+    size_t key_len_uncompressed_actual =
+        ephKey.toECPoint(openssl::EllipticCurvePointConversionForm::uncompressed).size();
+    size_t key_len_uncompressed_calculated =
+        getEcPublicKeyOctetLength(key, openssl::EllipticCurvePointConversionForm::uncompressed);
+    EXPECT_EQ(key_len_uncompressed_actual, key_len_uncompressed_calculated);
+}
+
+TEST_F(ECIESTests, testEncodedKeyLength)
+{
+    auto encCtx = encBuilder.buildEncryptionCtx(secp384PublicKey);
+    auto ephKey = encCtx->getEphemeralKey();
+    testGetEcPublicKeyOctetLength(ephKey, secp384Key);
+}
+
+TEST_F(ECIESTests, testEncodedKeyLengthNotFactor8)
+{
+    AsymmetricKeypair secp521Key = AsymmetricKeypair::generate(ECCSpec(openssl::ellipticCurveNid::SECP_521r1));
+    AsymmetricPublicKey secp521PublicKey = secp521Key;
+    auto encCtx = encBuilder.buildEncryptionCtx(secp521PublicKey);
+    auto ephKey = encCtx->getEphemeralKey();
+    testGetEcPublicKeyOctetLength(ephKey, secp521Key);
+}
+
```

Comment:
Can you please create one "static" test using the data in preparetestDataForECIESTests?

Reply:
The method is now moved to the key class and the test is now ensuring that the lengths match the result of the toECPoint method for all possible cases, so another test is probably superfluous at this point